src/robothub/replay_camera.py

```diff
@@ -0,0 +1,542 @@
+    """
+
+    def __init__(self, path: pathlib.Path):
+        image_files = [
+            os.path.join(str(path), f) for f in os.listdir(str(path)) if f.endswith((".png", ".jpg", ".jpeg"))
+        ]
+        image_files.sort()
+        self.image_files = image_files
+        self.current_frame = 0
+
+    def read(self) -> Tuple[bool, Optional[np.ndarray]]:
+        if self.current_frame < len(self.image_files):
+            frame = cv2.imread(self.image_files[self.current_frame])
+            self.current_frame += 1
+            return True, frame
+        else:
+            return False, None
+
+    def reset(self):
+        self.current_frame = 0
+
+    def isOpened(self):
+        return True
+
+    def release(self):
+        pass
+
+
+# from https://github.com/opencv/opencv/issues/21727#issuecomment-1068908078
+def BGR2YUV_NV12(src):
+    src_h, src_w = src.shape[:2]
+    # Convert BGR to YUV_I420
+    dst = cv2.cvtColor(src, cv2.COLOR_BGR2YUV_I420)
+    n_y = src_h * src_w
+    n_uv = n_y // 2
+    n_u = n_y // 4
+
+    # Extract the Y plane
+    y_plane = dst[:src_h].reshape((src_h, src_w))
+
+    # Extract the U and V planes, then interleave them for NV12 format
+    u_plane = dst[src_h : src_h + src_h // 4].reshape((-1, src_w // 2))
+    v_plane = dst[src_h + src_h // 4 :].reshape((-1, src_w // 2))
+
+    uv_plane = np.zeros((src_h // 2, src_w), dtype=np.uint8)
+    uv_plane[:, 0::2] = u_plane.reshape((-1, src_w // 2))
+    uv_plane[:, 1::2] = v_plane.reshape((-1, src_w // 2))
+
+    # Combine Y and interleaved UV planes into one NV12 image
+    nv12_img = np.vstack((y_plane, uv_plane)).astype(np.uint8)
+    return nv12_img
+
+
+def to_planar(arr: np.ndarray, shape: tuple) -> np.ndarray:
+    return cv2.resize(arr, shape).transpose(2, 0, 1).flatten()
+
+
+class ReplayCamera:
+    def __init__(self, pipeline: dai.Pipeline, fps: float, src: str, loop: bool = True):
+        # NOTE(miha): Replay node inputs/outputs
+        self._input_control: dai.Node.Input
+        self._input_config: dai.Node.Input
+        self._raw: dai.Node.Output
+        self._isp: dai.Node.Output
+        self._video: dai.Node.Output
+        self._still: dai.Node.Output
+        self._preview: dai.Node.Output
+        # self.frameEvent: dai.Node.Output
+        # self.InitialControl: dai.CameraControl
+
+        self._fps: float = fps
+        self._video_width: int = 1920
+        self._video_height: int = 1080
+        self._preview_width: int = 1280
+        self._preview_height: int = 720
+        self._color_order: dai.ColorCameraProperties.ColorOrder = dai.ColorCameraProperties.ColorOrder.BGR
+        self._interleaved = False
+        self._loop = loop
+
+        self._path_type: PathType
+        self._path = self._parse_src(src)
+        if self._path is None:
+            logging.error(f"Error in parsing src: {src}")
+            return
+
+        self.INPUT_CONTROL_STREAM_NAME = "input_control"
```

Comment:
I think those constants would be better defined as Enum, because you don't use them as class variables, you have to pass them into class methods as inputs